--skip-npm flag to Skip putting emsdk's npm on PATH

shlomif opened the pull request:
Fixes: #1142

shlomif committed:
skip-npm

https://github.com/emscripten-core/emsdk/issues/1048

shlomif committed:
cleanup

shlomif committed:
Refactoring / code cleanup.

See:

* https://en.wikipedia.org/wiki/Code_refactoring

* https://www.refactoring.com/

* https://www.joelonsoftware.com/2002/01/23/rub-a-dub-dub/

Some small optimisations may have slipped in as well.

sbc100 reviewed (commented):
I don't know if this patch aligns with the approch that we discuessed in the bug, but if we do take this approach I think you would just want to modify `get_required_path`.   We still want to install and activate our own version of node, we just don't want to add it to the users path, right?

shlomif committed:
Rename to skip_node and cleanup.

Per the comments on the pullreq

shlomif commented:
@sbc100 : thx. i fixed some low-hanging fruit

robbertvanginkel committed:
Add standalone_wasm feature to bazel emscripten_toolchain (#1145)

sbc100 committed:
3.1.28 (#1149)

jfirebaugh committed:
Upgrade to rules_nodejs 5.8.0 (#1150)

Fixes https://github.com/emscripten-core/emsdk/issues/1020

shlomif committed:
refactor

sbc100 committed:
3.1.29 (#1160)

jfirebaugh committed:
Pin Windows CI to Bazel 5.4.0 (#1163)

sbc100 committed:
Remove reference to fastcomp-latest. NFC (#1164)

fastcomp can only be install using explicit versions names so this name
doesn't work.

sbc100 committed:
Remove fastcomp SDK and fastcomp build rules. NFC (#1165)

Folks that want to work with fastcomp will now need to use an older
checkout of emsdk.

dschuff committed:
3.1.30 (#1167)

sbc100 committed:
Remove `-upstream` component of SDK names (#1166)

This name existed to distinguish the SDK from fastcomp, but as of #1165,
we no longer support fastcomp.

sbc100 committed:
Remove uneeded str. NFC (#1170)

These function already returns a string.

sbc100 committed:
Update README.md after #1166. NFC (#1172)

sbc100 committed:
3.1.31 (#1176)

juj committed:
Add support for Visual Studio 2022 (#1177)

* Add support for Visual Studio 2022 and migrate to using cmake --build when building on Windows. Leverage the VS2019 MSBuild 'Multi-ToolTask' feature CL_MPCount to saturate project builds properly to 100% CPU usage so building LLVM builds different cpp files in parallel. Clean up some code duplication around Visual Studio support.

* flake

* Work around Linux bot not having 'cmake --build . -j' flag.

juj committed:
Fix args ordering in LLVM build from previous commit (#1179)

juj committed:
Remove support for building with Visual Studio 2017. Since Jan 29, 2022, upstream LLVM has required VS2019 or VS2022 to build. So it has not been possible to build emsdk from source with VS2017 for a year. https://github.com/llvm/llvm-project/commit/058c5dfc78cd1a1a6075bba9799e63f3ec871c0d (#1178)

sbc100 committed:
Improve ordering in `emsdk list` (#1180)

We have an existing `version_key` helper function for sorting versions.

It also does a better job, producing output like:

```
All recent (non-legacy) installable versions are:
         3.1.31
         3.1.31-asserts
         3.1.30
         3.1.30-asserts
         3.1.29
         3.1.29-asserts
```

Rather than:

```
All recent (non-legacy) installable versions are:
         3.1.31
         3.1.30
         3.1.29
         3.1.28
         3.1.27
```

(with -assert versions listed after 3.1.0)

Blackhex committed:
Add Windows Arm64 support for compiling from source (#1186)

Blackhex committed:
Remove unused build_target_platform parameter from make_build (#1187)

sbc100 committed:
3.1.32 (#1188)

sbc100 committed:
Update some stale information in README.md (#1190)

- Remove reference to `~/.emscripten`.  We no longer use the home
  directory to store config information by default, either in
  emscripten or in emsdk.
- Remove some references to `The Emscripten Command Prompt`.  While I
  suppose this is referring to the windows-only `emcmdprompt.bat`, and I
  suppose it means "any shell where `activate` has been run", I think
  its more clear to simply avoid using the term.

kon72 committed:
Add missing binaries for bazel (#1082)

sbc100 committed:
Only add node/bin to user's PATH if one is not already found (#1189)

If the user already has a version of node in their PATH don't clobber
it.  This doesn't effect emscripten since the version of node we use
there is controlled via the config file, not via PATH.

Part of fix for #705.

tlively committed:
3.1.33 (#1200)

sbc100 committed:
v3.1.34 (#1203)

sbc100 committed:
Update arm64 linux release (#1204)

Fixes: #1202

robbertvanginkel committed:
Add linux/arm64 support to bazel rules (#1156)

* Add linux/arm64 support to bazel rules

* Make linux arm64 sha optional and PR feedback

Not all releases have linux shas. Use getattr to default to no sha if none was
present. This way non linux arm64 builds should not fail.

* Update linux arm64 sha for 3.1.33

This is the latest arm64 release according to #1204. There do not seem to be a any releases between .21 and .33.

dschuff committed:
3.1.35 (#1208)

sbc100 committed:
Add `activated_path_skip` to node v14.18.2 packages (#1209)

This should have been part of #1189.  See #1183.

sbc100 committed:
Update node version 14.15.5 -> 15.14.0 (#829)

sbc100 committed:
3.1.36 (#1210)

sbc100 committed:
3.1.37 (#1212)

sbc100 committed:
3.1.38 (#1215)

dschuff committed:
3.1.39 (#1217)

dschuff committed:
3.1.40 (#1221)

dschuff committed:
Update mac builders to use Circle's gen2 mac runner (#1222)

This runner is faster and more efficient.
Also factor the mac configuration into an executor.

pnacht committed:
Add security policy (#1224)

Signed-off-by: Pedro Kaj Kjellerup Nacht <pnacht@google.com>

kleisauke committed:
Docker: remove remnant of fastcomp (#1226)

The fastcomp backend was removed in Emscripten v2.0.0.

kleisauke committed:
Docker: remove redundant `EMSDK_NODE` env (#1225)

sbc100 committed:
3.1.41 (#1230)

dschuff committed:
Update node, 15.14.0 -> 16.20.0 (#1232)

This allows us to use the native ARM64 version on MacOS.
Also update the test scripts to work on ARM64 mac, and skip tests that aren't relevant.

sbc100 committed:
3.1.42 (#1238)

sbc100 committed:
Prefer `arm64` over `aarch64` internally. NFC (#1246)

This seems like more commonly known/used name for the architecture
these days.

sbc100 committed:
Add some basic testing of arm64 linux (#1247)

sbc100 committed:
Add a warning when trying to install linux-arm64 binaries (#1249)

See #547

sbc100 committed:
Fix `activated_path_skip` on repeated running of `emsdk_env.sh` (#1250)

The first time around `node` was being correctly added to the PATH, but
the second time around this code was observing the emsdk copy of node
in the PATH and assuming it could be skipped.

Fixes: #1240

kripken committed:
Fix a typo (#1251)

kleisauke committed:
Docker: use a entrypoint script (#1227)

In favor of `/etc/bash.bashrc`, which is only read for interactive
shells.

sbc100 committed:
3.1.43 (#1254)

sbc100 committed:
Have create_release.py push the new branch automatically (#1255)

Having used this script for a while now I'm not sure there is any
point if leaving this last step as a manual push.

sbc100 committed:
Update linux arm64 to 3.1.41 (#1256)

sbc100 committed:
Rename `zipfile` to `archive` or `download` as appropriate. NFC (#1257)

sbc100 committed:
Update linux arm64 to 3.1.43 (#1258)

sbc100 committed:
3.1.44 (#1260)

martijneken committed:
wasm_cc_binary: Specify a default OS. Allow users to override platform. (#1262)

* wasm_cc_binary: Specify a default OS. Allow users to override platform.

Problem: https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/pull/157#discussion_r1277763118

This is solving the problem in two different ways. Please let me know
your thoughts about both approaches, as either will work.

Signed-off-by: Martijn Stevenson <mstevenson@google.com>

* Rework platform selection to trigger os:wasi off standalone attr

Signed-off-by: Martijn Stevenson <mstevenson@google.com>

---------

Signed-off-by: Martijn Stevenson <mstevenson@google.com>

mattsoulanille committed:
Add starlark highlighting to Bazel readme (#1264)

Mark the build files as using the starlark language so they have the correct syntax highlighting.

dschuff committed:
3.1.45 (#1269)

Also update emscripten include dir to v18, and change 17 to wildcard in emscripten deps

sbc100 committed:
Update linux arm64 build. NFC (#1270)

dschuff committed:
Fix the format of the .flake8 file, and exclude the script directory from the py2 linter (#1272)

Flake8's INI config file format requires commas after each line. Because our file didn't have them, the exclude list
wasn't set up correctly, and the --extend-exclude flag wasn't working. This PR fixes the .flake8 file.

Also, update flake8 to the latest version available (because version 3.8 is required to get the --extend-exclude flag)
and use the flag to exclude the files in the scripts/ directory from the python2 linter (since the scripts are python3).

dschuff committed:
Replace update_bazel_workspace.sh with a python script (#1271)

This script is (IMO) more readable, but the real reason for this change is that
it raises an error message when the binary package fails to download. (The shell
script silently generated a bogus hash instead, because the shell's `set -e`
builtin does not affect commands executing inside a $() context.
It seemed just as easy to rewrite the script in Python as to fix that.

This change also updates some outdated filename references.

walkingeyerobot committed:
[bazel] populate all_files (#1274)

Fixes #1273. This was broken by #1045 with the comment "* all_files not needed?" They were needed.

sbc100 committed:
3.1.46 (#1279)

sbc100 committed:
Switch to .xz by default for SDK downloads (#1281)

This is a bit of a hack but I can't think of another way to do it.
Basically when downloading SDKs, we first try the new `.xz` extension.
If that fails, we fall back to the old `.tbz2`.  Both these first two
download attempts we run in "silent" mode.  If both of them fail we
re-run the original request in non-silent mode so that the error message
will always contain the original `.xz` extension.

See #1235

sbc100 commented:
I think this can be closed now that #1189 has landed emsdk will no longer add its node to the PATH if you already have one .
